### PR TITLE
Write changelog for upcoming release

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Initial draft for what the v0.5.0 (i.e. `v0.14.0` PS release-compatible) release notes should be.

Part of purescript/purescript#3985

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
I'm assuming we'll want a meta-issue to track which repos have had their changelog written.

**Comment by hdgarrood** [comment]
I've just spotted that this repo has a v4.1.1 release which isn't mentioned in the changelog, that should probably be fixed. These release notes also don't mention #229, are you still working on them?

**Comment by JordanMartinez** [comment]
No, I didn't include #229 because I believe we reverted that change in #233 / #232.

I'll add the 4.1.1 release. It wasn't included because it was just a tag, not an official GH release.

**Comment by JordanMartinez** [comment]
Should we remove [`unsafeCompare`](https://github.com/purescript/purescript-prelude/commit/d84c89f478f2e9b4a13847b236f0e4cd4a5783b9)? It was deprecated in `v4.1.1`.

**Comment by hdgarrood** [comment]
We reverted the change to `ap` but we did still move the Applicative superclass law from Monad to Bind.

**Comment by JordanMartinez** [comment]
Ah... Gotcha. I've added that to this PR.

File: CHANGELOG.md
Comment (milesfrain) [inline CHANGELOG.md]: I don't know if this level of detail (where each PR is mentioned) is necessary for changelogs. It would probably be fine to just write something like "Added more examples" or not even mention that, since it doesn't really affect existing users of the library.
I think the main purpose of the changelog is to help folks quickly review API changes since their last update.

Comment (JordanMartinez) [reply CHANGELOG.md]: I'm good with making this change if others agree. I thought "Added more docs" or something wasn't that helpful either.

Comment (thomashoneyman) [reply CHANGELOG.md]: I think it's reasonable to point out documentation updates, but I agree they ought to be as concise as possible. For example:

```
Other improvements:
- Added examples to documentation comments for `lift2`, `const`, and `power` (#213, #214, #253)
- Improved documentation for `do` notation for `Array`, the purpose of the `Monoid` and `Semigroup` newtypes, and the representation of `Unit` in FFI code (#217, #223)
```

Comment (thomashoneyman) [reply CHANGELOG.md]: (They're already de-prioritized by being bundled under the "Other improvements" section, which is easier to skip if you're only skimming for API changes)

**Review by thomashoneyman — APPROVED** [review]
I think others should have a chance to read this changelog and weigh in on its contents, so we should wait for a second review before merging. That said, a 👍 as far as I'm concerned and thank you for getting this started.

File: CHANGELOG.md
Comment (hdgarrood) [reply CHANGELOG.md]: I disagree, I think almost every PR should be referenced in the changelog. It's not as if there's a massive amount of detail to wade through here, and it's important that we provide credit to contributors for the changes of theirs we've merged.

File: CHANGELOG.md
Comment (hdgarrood) [inline CHANGELOG.md]: The phrase "do notation" isn't code, so we shouldn't use backticks for it here.

File: CHANGELOG.md
Comment (hdgarrood) [inline CHANGELOG.md]: I would suggest phrasing this as "Support compiler version `v0.14.0`, and drop support for previous versions", as it's a bit more clear that `v0.14.0` refers to the compiler version that way as opposed to the version of some other tool, and also it means that it's clear that this version of the library can't be used with older compilers.

File: CHANGELOG.md
Comment (hdgarrood) [reply CHANGELOG.md]: Personally I think having separate bullet points for each of the separate documentation PRs (as you do currently) is ideal. We can group them all under a "Documentation improvements" heading so that people who aren't interested in anything that doesn't affect the actual API can easily skim past them.

**Comment by hdgarrood** [comment]
Let's remove unsafeCompare, it has already been deprecated for over a year.

File: CHANGELOG.md
Comment (thomashoneyman) [reply CHANGELOG.md]: I agree almost all pull requests should be referenced in the changelog. I'm also fine with bullet points for each documentation PR.

As a note, the more concise version I added doesn't drop any PRs, it just allows some of them to be included in the same line if they're logically related. I also think _that's_ fine, but we should probably have an agreed-upon default.

**Comment by JordanMartinez** [comment]
@hdgarrood I've addressed the feedback you raised.

**Comment by JordanMartinez** [comment]
Any other feedback? Or am I clear to merge this?

File: CHANGELOG.md
Comment (hdgarrood) [inline CHANGELOG.md]: I think having a bullet point labelled "improve documentation in various things" under a parent "Documentation improvements" is a bit redundant. How about:
```suggestion
- Clarify `Array`'s do notation and the purposes of `Monoid` and `Semigroup` newtypes (#217)
```

**Comment by hdgarrood** [comment]
I would like to go over the commits since 4.1.1 just to make sure we haven't missed anything out, but other than that, this looks good to me. Thanks!

**Comment by hdgarrood** [comment]
I think the following PRs should be referenced in this changelog: #220, #227, #238

**Comment by JordanMartinez** [comment]
Added those PRs. How about now?

File: CHANGELOG.md
Comment (hdgarrood) [inline CHANGELOG.md]: Oh, last thing, maybe it would be good to leave this stuff under "Unreleased" for now so that we don't forget to update the placeholder link and date? I don't really mind though.

**Review by hdgarrood — APPROVED** [review]

File: CHANGELOG.md
Comment (JordanMartinez) [reply CHANGELOG.md]: Makes sense to me. I'll add that change and then merge.

**Pull request merged**
